src/main/java/seedu/address/model/friend/Friend.java

```diff
@@ -0,0 +1,94 @@
+package seedu.address.model.friend;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents a Friend in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Friend {
+
+    // Identity fields
+    private final seedu.address.model.friend.Name name;
+    private final seedu.address.model.friend.Passport passport;
```

Comment:
According to @jiaweiteo , only the name should be considered as identity fields, while passport and phone should be data fields